elytron/src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -601,19 +631,68 @@ static ResourceDefinition getTrustManagerDefinition() {
+                if (crlCertPath != null) {
+                    ROOT_LOGGER.warn("maximum-cert-path in certificate-revocation-list is for legacy support. Please use only the one in trust-manager!");
+                    if (maxCertPath != null) {
+                        throw new OperationFailedException("Multiple maximum-cert-path definitions found.");
+                    }
+                    maxCertPath = crlCertPath;
+                }
+
+                String crlPath = null;
+                String crlRelativeTo = null;
                 final InjectedValue<PathManager> pathManagerInjector = new InjectedValue<>();
 
-                if (crlPath != null) {
-                    if (crlRelativeTo != null) {
-                        serviceBuilder.addDependency(PathManagerService.SERVICE_NAME, PathManager.class, pathManagerInjector);
-                        serviceBuilder.requires(pathName(crlRelativeTo));
+                if (crlNode.isDefined()) {
+                    crlPath = PATH.resolveModelAttribute(context, crlNode).asStringOrNull();
+                    crlRelativeTo = RELATIVE_TO.resolveModelAttribute(context, crlNode).asStringOrNull();
+
+                    if (crlPath != null) {
+                        if (crlRelativeTo != null) {
+                            serviceBuilder.addDependency(PathManagerService.SERVICE_NAME, PathManager.class, pathManagerInjector);
+                            serviceBuilder.requires(pathName(crlRelativeTo));
+                        }
+                    }
+                }
+                boolean preferCrls = PREFER_CRLS.resolveModelAttribute(context, ocspNode).asBoolean(false);
+                String responder = RESPONDER.resolveModelAttribute(context, ocspNode).asStringOrNull();
+                URI responderUri;
+                try {
+                    responderUri = responder == null ? null : new URI(responder);
+                } catch (Exception e) {
+                    throw new OperationFailedException(e);
+                }
+
+                X509RevocationTrustManager.Builder builder = X509RevocationTrustManager.builder();
+                builder.setResponderURI(responderUri);
+                if (maxCertPath != null) {
+                    builder.setMaxCertPath(maxCertPath.intValue());
+                }
+                if (crlNode.isDefined()) {
+                    if (!ocspNode.isDefined()) {
+                        builder.setPreferCrls(true);
+                        builder.setNoFallback(true);
+                    }
+                }
+                if (ocspNode.isDefined()) {
+                    builder.setResponderURI(responderUri);
+                    if (!crlNode.isDefined()) {
+                        builder.setPreferCrls(false);
+                        builder.setNoFallback(true);
+                    } else {
+                        builder.setPreferCrls(preferCrls);
                     }
                 }
```

Comment:
I don't see any `setSoftFail()` method call - has it been missed?

Reply:
Fixed